cpp/src/arrow/ipc/writer.cc

```diff
@@ -810,6 +815,25 @@ class SparseTensorSerializer {
     return Status::OK();
   }
 
+  Status VisitSparseCSFIndex(const SparseCSFIndex& sparse_index) {
+    std::shared_ptr<Buffer> indptrs, indices;
+    std::vector<std::shared_ptr<Buffer>> indptr_buffers, indices_buffers;
+
+    for (const std::shared_ptr<arrow::Tensor> indptr : sparse_index.indptr()) {
+      indptr_buffers.emplace_back(indptr->data());
+    }
+    for (const std::shared_ptr<arrow::Tensor> indices : sparse_index.indices()) {
+      indices_buffers.emplace_back(indices->data());
+    }
+
+    RETURN_NOT_OK(ConcatenateBuffers(indptr_buffers, default_memory_pool(), &indptrs));
```

Comment:
@mrkn @pitrou do you know of a better approach to do this given the [schema](https://github.com/apache/arrow/blob/7e325a24c25575f83095afe84413beda4e05a64a/format/SparseTensor.fbs#L160)?

Reply:
I'm not sure I understand. `SparseTensorIndexCSF.indptrBuffers` and `SparseTensorIndexCSF.indicesBuffers` are supposed to be arrays of buffers. Why do you need to concatenate them?

Reply:
My understanding is that [`Assemble`](https://github.com/apache/arrow/blob/7e325a24c25575f83095afe84413beda4e05a64a/cpp/src/arrow/ipc/writer.cc#L774) expects a vector of buffers to serialize.
Alternatively we could do:
```
    for (const std::shared_ptr<arrow::Tensor> indptr : sparse_index.indptr()) {
      out_->body_buffers.emplace_back(indptr->data());
    }
    for (const std::shared_ptr<arrow::Tensor> indices : sparse_index.indices()) {
      out_->body_buffers.emplace_back(indices->data());
    }
```

Reply:
That would look better, but you also have to fill the `SparseTensorIndexCSF` fields appropriately.

By the way, I think you should tackle Python serialization in a separate PR and focus on IPC here.

Reply:
> By the way, I think you should tackle Python serialization in a separate PR and focus on IPC here.

Will do.